libbeat/outputs/otelconsumer/otelconsumer.go

```diff
@@ -92,6 +97,17 @@ func (out *otelConsumer) logsPublish(ctx context.Context, batch publisher.Batch)
 		meta["version"] = out.beatInfo.Version
 		meta["type"] = "_doc"
 
+		if id, ok := meta["_id"]; ok {
+			// Set a log record attribute for the document ID, this is used for deduplication
+			// in the elasticsearchexporter.
+			//
+			// See https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36882.
+			switch id := id.(type) {
+			case string:
+				logRecord.Attributes().PutStr(esDocumentIDAttribute, id)
```

Comment:
Feels a bit weird to always set this attribute, but I'm not aware of a way to test if the elasticsearchexporter is part of the collector instance in order to enable this. Either way this should be fine since log record attributes are not persisted into the final document when using the bodymap encoding.

Reply:
If I understood it correctly, regardless of the final exporter, this field is always set but it does not get persisted in the final document, is that it?

Could you add this explanation to the code as well?